webapi/src/main/java/org/sakaiproject/webapi/controllers/SitesController.java

```diff
@@ -85,4 +93,88 @@
 
         return data;
 	}
+
+    @Resource
+    private SecurityService securityService = (SecurityService) ComponentManager.get(SecurityService.class);
+
+    @Resource
+    private PreferencesService preferencesService = (PreferencesService) ComponentManager.get(PreferencesService.class);
+
+    @Resource
+    private ActiveToolManager activeToolManager = (ActiveToolManager) ComponentManager.get(ActiveToolManager.class);
+
+    @Resource
+    private ToolManager toolManager = (ToolManager) ComponentManager.get(ToolManager.class.getName());
+
+	@GetMapping(value = "/users/{userId}/favs", produces = MediaType.APPLICATION_JSON_VALUE)
+    public Map<String, List<Map<String, Object>>> getFavsWithPages(@PathVariable String userId)
+        throws UserNotDefinedException {
+
+        Session session = checkSakaiSession();
+        String reqUser = session.getUserId();
+        //If non admin user requests data of another user, return empty map and warn
+        if(!securityService.isSuperUser() && !userId.equals(reqUser)) {
+            log.warn("User " + reqUser + " requested favorites from user " + userId);
+            return new HashMap<>();
+        }
+        return getUserFavoriteSitesWithPages(userId);
+	}
+
+	@GetMapping(value = "/user/favs", produces = MediaType.APPLICATION_JSON_VALUE)
```

Comment:
this mapping should probably be `/sites/user/favorites` see the other controllers and lets make sure we get a consensus on what the url should be here?

Reply:
/users/{userId}/favorites and /users/favorites for consistency with other controllers.

Reply:
Yes, there is also /users/{userId}/sites in the same controller. Thought its conflicting at first, but /users/{userId}/favorites and /users/favorites works fine